### DISCUSSION TITLE
Setting up test server - Error unknown certificate type

### DISCUSSION THREAD

**Discussion by foggyoutside** [open-quantum-safe/oqs-provider#508] [Q&A] [answered]
I am on a up to date Ubuntu VM with openssl 3.3.1 installed. Entering "openssl list -providers" shows both the default and the oqs-provider. Both are activated in the openssl.cnf file as explained in the USAGE.md.

Creating keys and certificates as described in USAGE.md works, but trying to set up the test server ends up in a error message (see picture). I know there are premade docker images but I would like to set it up myself and experiment with its compatibilty with stunnel.
![Screenshot_vm](https://github.com/user-attachments/assets/9dbc4713-30bb-456b-8b24-f8077f9338e2)
the same command but running with sudo, different result:
![image](https://github.com/user-attachments/assets/cd4eb00c-5104-4679-bf25-ee1307aa14c8)

here the output from listing the providers
![image](https://github.com/user-attachments/assets/0bffed3d-6488-408d-9a3b-c9bfd72fe787)

my setting in the openssl.cnf. Could it be a problem with the "module"? But if I comment out that line I receive another error (last screenshot).
![image](https://github.com/user-attachments/assets/825d3c84-96a9-43ec-a85f-200ba851b0d1)
![image](https://github.com/user-attachments/assets/dbebac70-5222-4c58-9ab3-dc3f3cd6c794)

the permissions on the key/certificates seem right to me:
![image](https://github.com/user-attachments/assets/0fe3f675-7d0b-41c5-85a5-fbe4f8abff54)

**Comment by baentsch** [comment]
The error message seems to indicate that oqsprovider didn't properly load. You may want to try using OPENSSL_MODULES environment variable instead.

**Comment by foggyoutside** [comment, marked as answer]
I couldn't figure it out. Reinstalled everything using the following script https://www.linkedin.com/pulse/quantum-safing-ubuntu-debian-igor-barshteyn/ 

Works perfectly now. The OQS-Provider installation docu wasn't intuitive.
